esp-mbedtls/src/lib.rs

```diff
@@ -244,12 +244,133 @@ impl<'a> X509<'a> {
     }
 }
 
-/// Certificates used for a connection.
-///
-/// # Note:
-/// Both [certificate](Certificates::certificate) and [private_key](Certificates::private_key) must be set in pair.
-#[derive(Debug, Clone, Copy, PartialEq)]
-pub struct Certificates<'a> {
+/// Creates a wrapper over [mbedtls_x509_crt] to safely manage allocation and freeing on drop
+#[derive(Debug)]
+struct MbedTLSX509Crt(*mut mbedtls_x509_crt);
+
+impl MbedTLSX509Crt {
+
+    fn new(certificate: X509<'_>) -> Result<Self, TlsError> {
```

Comment:
To fix the "TODO" inside this methods, we actually need two methods:

A copying variant, for PEM and perhaps for DER too where the user would not like to keep a ref to the `x509` contents:
```rust
impl MbedTlsX509Crt<'static> {
    fn new(x509: X509<'_>) -> Result<Self, TlsError> { ... }
    ...
```

A non-copying variant (would only work for DER X509:
```rust
impl<'d> MbedTlsX509Crt<'d> {
    fn new(x509: X509<'d>) -> Result<Self, TlsError> { ... }
    ...
}
```